presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -84,6 +94,48 @@ public FilterNode filter(Expression predicate, PlanNode source)
         return new FilterNode(idAllocator.getNextId(), source, predicate);
     }
 
+    public TableScanNode tableScan(List<Symbol> symbols, Map<Symbol, ColumnHandle> assignments)
+    {
+        Expression originalConstraint = null;
+        return new TableScanNode(
+                idAllocator.getNextId(),
+                new TableHandle(
+                        new ConnectorId("testConnector"),
+                        new TestingTableHandle()),
+                symbols,
+                assignments,
+                Optional.empty(),
+                TupleDomain.all(),
+                originalConstraint
+        );
+    }
+
+    public TableFinishNode tableDelete(SchemaTableName schemaTableName, PlanNode deleteSource, Symbol deleteRowId)
+    {
+        TableWriterNode.DeleteHandle deleteHandle = new TableWriterNode.DeleteHandle(
+                new TableHandle(
+                        new ConnectorId("testConnector"),
+                        new TestingTableHandle()),
+                schemaTableName
+        );
+        return new TableFinishNode(
+                idAllocator.getNextId(),
+                exchange(e -> e
```

Comment:
I am not aiming at a generic factory method for table finish nodes here. That's why this method is called `tableDelete`.